src/features/authentication/registration/ui/DetailForm.tsx

```diff
@@ -24,9 +26,22 @@ export function DetailForm(): ReactElement {
 
     const dispatch = useAppDispatch();
 
-    const onSubmit = (data: UserDataValidationSchema) => {
-      // TODO: Check email emailRef.current?.value
-      dispatch(userDataFilled({username: data.username, email: data.email}))
+    const [
+      checkEmailTrigger,
+    ] = useCheckEmailMutation()
+
+    const onSubmit = ({username, email}: UserDataValidationSchema) => {
+      // setFormValuesInStorage(AUTH_FORM_VALUES_KEY, { email });
+      checkEmailTrigger({email})
+          .unwrap()
+          .then((payload) => {
+            if (payload) {
+              setError('email', {message: "This email address is already in use"});
```

Comment:
I suggest to write error message in russian because our users are russian speaking.